hepcrawl/parsers/crossref.py

```diff
@@ -0,0 +1,387 @@
+        """
+
+        self.builder.add_abstract(self.abstract)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        for reference in self.references:
+            self.builder.add_reference(reference)
+        self.builder.add_imprint_date(self.imprints)
+        for author in self.authors:
+            self.builder.add_author(author)
+        for license_instance in self.license:
+            self.builder.add_license(**license_instance)
+        self.builder.add_url(self.url)
+        self.builder.add_publication_info(**self.publication_info)
+        self.builder.add_number_of_pages(self.number_of_pages)
+        self.builder.add_title(self.title, subtitle=self.subtitle)
+        self.builder.add_document_type(self.document_type)
+
+        return self.builder.record
+
+    @property
+    def document_type(self):
+        doc_type = self.record.get("type")
+        return DOC_TYPE_MAP[doc_type]
+
+    @property
+    def title(self):
+        title = self.record.get("title")
+
+        if title:
+            return title[0]
+        else:
+            return None
+
+    @property
+    def subtitle(self):
+        subtitle = self.record.get("subtitle")
+
+        if subtitle:
+            return subtitle[0]
+        else:
+            return None
+
+    @property
+    def dois(self):
+        value = self.record.get("DOI")
+        dois = [
+            {'doi': value, 'material': self.material}
+        ]
+
+        return dois
+
+    @property
+    def url(self):
+        return self.record.get("URL")
+
+    @property
+    def material_source(self):
+        return self.record.get("source")
+
+    @property
+    def material(self):
+        title = self.title
+        subtitle = self.subtitle
+        if 'erratum' in title.lower() or 'erratum' in subtitle.lower():
+            material = 'erratum'
+        elif 'addendum' in title.lower() or 'addendum' in subtitle.lower():
+            material = 'addendum'
+        elif "publisher's note" in title.lower() or "publisher's note" in subtitle.lower():
+            material = 'addendum'
+        else:
+            material = 'publication'
+
+        return material
+
+    @property
+    def publication_info(self):
+        publication_info = {
+            'artid': self.artid,
+            'journal_title': self.journal_title,
+            'journal_issue': self.journal_issue,
+            'journal_volume': self.journal_volume,
+            'page_start': self.page_start,
+            'page_end': self.page_end,
+            'year': self.year,
+            'material': self.material,
+            'parent_isbn': self.parent_isbn,
+        }
+
+        return publication_info
+
+    @property
+    def parent_isbn(self):
+        isbn = self.record.get("ISBN")
+        if isbn:
+            return isbn[0]
+        else:
+            return isbn
+
+    @property
+    def journal_title(self):
+        j_title = self.record.get("short-container-title")
```

Comment:
it's probably better to use the full title here. This will be shortened later according to Inspire's convention which might be different from the publisher's.